### PR TITLE
Updating Servant-client library to use new ClientM (it is now a ReaderT)

### PR DESCRIPTION
**Pull request by sebashack**
In this PR we are just updating servant-client and the ClientM monad so that it fits the new implementation which carries a ReaderT.

### CODE REVIEW

File: Web/Harvest/API.hs [hunk 1]
```diff
@@ -65,35 +66,32 @@ getUsers_ :<|> getTimeEntries_ = client (Proxy :: Proxy HarvestAPI)
...
-runHarvestQuery :: MonadIO m
-  => (BasicAuthData -> Manager -> BaseUrl -> ClientM a) -- ^ Query function
-  -> Manager           -- ^ HTTPS Manager
-  -> Credentials       -- ^ Credentials
-  -> m (Either ServantError a) -- ^ The result
-runHarvestQuery action manager Credentials {..} = liftIO $ do
-  let host     = BC8.unpack credentialsAccount ++ ".harvestapp.com"
-      authData = BasicAuthData credentialsUsername credentialsPassword
-  runExceptT (action authData manager (BaseUrl Https host 443 ""))
+runHarvestQuery :: (BasicAuthData -> ClientM a) -- ^ Query function
```
Comment (juanpaucar) [inline Web/Harvest/API.hs]: Could we replace it by its type synonym?
`type Query a = BasicAuthData -> ClientM a`

File: Web/Harvest/API.hs [hunk 2]
```diff
@@ -65,35 +66,32 @@ getUsers_ :<|> getTimeEntries_ = client (Proxy :: Proxy HarvestAPI)
 
 -- | Get list of all users for specific account.
 
-getUsers :: MonadIO m
-  => Manager           -- ^ HTTPS manager
-  -> Credentials       -- ^ Credentials
-  -> m (Either ServantError [User]) -- ^ Result of request
-getUsers = runHarvestQuery getUsers_
+getUsers :: Manager                         -- ^ Http Manager
```
Comment (juanpaucar) [inline Web/Harvest/API.hs]: `HTTP` is an acronym let's keep it in uppercase

File: Web/Harvest/API.hs [hunk 3]
```diff
@@ -65,35 +66,32 @@ getUsers_ :<|> getTimeEntries_ = client (Proxy :: Proxy HarvestAPI)
...
 
 -- | Get time entries for specific date and user.
 
-getTimeEntries :: MonadIO m
-  => Manager           -- ^ HTTPS manager
-  -> Credentials       -- ^ Credentials
-  -> Day               -- ^ Date of interest
-  -> UserId            -- ^ User id
-  -> m (Either ServantError TimeEntries)
+getTimeEntries :: Manager           -- ^ Http Manager
```
Comment (juanpaucar) [inline Web/Harvest/API.hs]: same here, please

File: stack.yaml [hunk 4]
```diff
@@ -1,7 +1,3 @@
-resolver: lts-5.17
+resolver: lts-8.22
```
Comment (juanpaucar) [inline stack.yaml]: not very important but, lts-8.23 is already available

**Review by juanpaucar — COMMENTED** [review]
Just some minor comments, the overall looks very good

File: Web/Harvest/API.hs (hunk 2 above)
Comment (sebashack) [reply Web/Harvest/API.hs]: cool

Comment (sebashack) [reply Web/Harvest/API.hs]: cool

File: Web/Harvest/API.hs (hunk 3 above)
Comment (sebashack) [reply Web/Harvest/API.hs]: cool

File: Web/Harvest/API.hs (hunk 1 above)
Comment (sebashack) [reply Web/Harvest/API.hs]: Sure

File: stack.yaml (hunk 4 above)
Comment (sebashack) [reply stack.yaml]: Crazy!

**Review by juanpaucar — APPROVED** [review]
:shipit:

**Pull request merged**